ast/desugar/Desugar.cc

```diff
@@ -454,6 +454,21 @@ void desugarPatternMatchingVars(InsSeq::STATS_store &vars, DesugarContext dctx,
     }
 }
 
+// Desugar `in` and `=>` online pattern matching
+ExpressionPtr desugarOnelinePattern(DesugarContext dctx, core::LocOffsets loc, unique_ptr<parser::Node> &match) {
+    auto matchExpr = MK::RaiseUnimplemented(loc);
+    auto bodyExpr = MK::RaiseUnimplemented(loc);
+    auto elseExpr = MK::EmptyTree();
+
+    InsSeq::STATS_store vars;
+    desugarPatternMatchingVars(vars, dctx, match);
+    if (!vars.empty()) {
+        bodyExpr = MK::InsSeq(match->loc, std::move(vars), std::move(bodyExpr));
+    }
+
+    return MK::If(loc, std::move(matchExpr), std::move(bodyExpr), std::move(elseExpr));
```

Comment:
Is this emitting an `if` where the condition is `MK::RaiseUnimplemented` because we don't have a way of saying whether the pattern matched yet?

Reply:
The idea was to allow these constructs to parse while doing as little as possible until the feature stabilizes upstream.

Once we have a better idea of what we want to do with pattern matching regarding typing, we can implement all these calls.